net/SessionManager.c

```diff
@@ -261,8 +261,9 @@ static Iface_DEFUN incomingFromSwitchIf(struct Message* msg, struct Iface* iface
         uint8_t* herKey = caHeader->handshake.publicKey;
         uint8_t ip6[16];
         // a packet which claims to be "from us" causes problems
-        if (!AddressCalc_addressForPublicKey(ip6, herKey)) {
-            Log_debug(sm->log, "DROP Handshake with non-fc key");
+        AddressCalc_addressForPublicKey(ip6, herKey);
+        if (!AddressCalc_validAddress(ip6)) {
```

Comment:
If you're already going to be returning the return value of `AddressCalc_validAddress(addressOut)`, why then call `AddressCalc_validAddress` again in the if?